commit 19520bf7b1cde44ca798a7b534bdb38878255645 (refs/changes/65/2065/2)
Author: James Nugent <james@joyent.com>
Date:   2017-06-08T19:20:21-05:00 (2 years, 4 months ago)
    
    OS-6174 Use first NIC for admin network if admin_nic_autoselect=true

diff --git a/overlay/generic/lib/svc/method/net-physical b/overlay/generic/lib/svc/method/net-physical
index 7fbae783..683360d0 100644
--- a/overlay/generic/lib/svc/method/net-physical
+++ b/overlay/generic/lib/svc/method/net-physical
@@ -498,6 +498,18 @@ if smf_is_globalzone; then
     setup_mtu
 
     # Setup admin NIC
+
+    # If there is no NIC with the admin tag, and the config has
+    # admin_nic_autoselect=true, designate the first NIC reported
+    # by dladm for admin use. This is useful in environments where
+    # the NICs are known to change beneath us.
+    if [[ "${CONFIG_admin_nic_autoselect}" == "true" ]] \
+        && ! nictagadm exists admin ; then
+        nictagadm add admin "$(dladm show-phys -m -p -o address | head -n1)"
+        nictagadm list
+        SYSINFO_NIC_admin=$(dladm show-phys -m -p -o link | head -n1)
+    fi
+
     if [[ -z "${SYSINFO_NIC_admin}" ]]; then
         echo "ERROR: admin NIC not found, unable to bring up admin network."
         exit ${SMF_EXIT_ERR_CONFIG}
